config/module.go

```diff
@@ -97,3 +102,60 @@ func (m Module) Equals(other Module) bool {
 	//nolint:govet
 	return reflect.DeepEqual(m, other)
 }
+
+var tarballExtensionsRegexp = regexp.MustCompile(`\.(tgz|tar\.gz)$`)
+
+// NeedsSyntheticPackage returns true if this is a local module pointing at a tarball.
+func (m Module) NeedsSyntheticPackage() bool {
+	return m.Type == ModuleTypeLocal && tarballExtensionsRegexp.MatchString(strings.ToLower(m.ExePath))
+}
+
+// SyntheticPackage creates a fake package for a local module which points to a local tarball.
+func (m Module) SyntheticPackage() (PackageConfig, error) {
+	var ret PackageConfig
+	if m.Type != ModuleTypeLocal {
+		return ret, errors.New("non-local package passed to syntheticPackage")
+	}
+	ret.Name = fmt.Sprintf("synthetic-%s", m.Name)
+	ret.Package = ret.Name
+	ret.Type = PackageTypeModule
+	ret.LocalPath = m.ExePath
+	return ret, nil
+}
+
+// syntheticPackageExeDir returns the unpacked ExePath for local tarball modules.
+func (m Module) syntheticPackageExeDir() (string, error) {
+	pkg, err := m.SyntheticPackage()
+	if err != nil {
+		return "", err
+	}
+	return pkg.LocalDataDirectory(viamPackagesDir), nil
+}
+
+// EntrypointOnlyMetaJSON is a miniature version . We do this to avoid a circular dep between CLI and RDK.
+// Better option is to move meta.json definition to this config package.
+type EntrypointOnlyMetaJSON struct {
```

Comment:
if not going to do it in this pr, file a followup ticket to track this

Reply:
ticketed https://viam.atlassian.net/browse/RSDK-7594